### PR TITLE
WIP: Clear path_util usages

### LINKED ISSUE #5030: Clean up path_util compatibility wrapper
[Moving](https://github.com/Tribler/tribler/pull/5025) to `Pathlib` required creating a compatibility wrapper module `path_util` implementing some `os.path`-like functions. Cleaning it up will make the code cleaner and will reduce the probability of masking path-handling errors.

So, everyone is welcome to try removing `path_util.*` lines from the code and replace them with proper Pathlib methods!

### PR DESCRIPTION
**Pull request by Solomon1732**
Clear path_util usages. Closes #5030 
Edit: In addition some minor refractoring.

### CODE REVIEW

**Comment by Solomon1732** [comment]
There is a reference in the code to a `path_util.splitdrive` function which I can't find anywhere. What should I use instead?
https://github.com/Tribler/tribler/blob/bcf03e0ea696ce0c704426a5b4120bc5c675d6de/src/tribler-core/tribler_core/utilities/osutils.py#L122

**Comment by ichorid** [comment]
> There is a reference in the code to a `path_util.splitdrive` function which I can't find anywhere. What should I use instead?
> 
> https://github.com/Tribler/tribler/blob/bcf03e0ea696ce0c704426a5b4120bc5c675d6de/src/tribler-core/tribler_core/utilities/osutils.py#L122

The function `get_free_space` is not used anywhere in the code. It is completely dead :skull:  Just remove `get_free_space` from the codebase altogether. And don't hesitate to look for unused code in the future. As I had mentioned before, there is *a lot* of dead code in Tribler now. Some dead code even has tests for it, but is actually not used anywhere else! So, don't take anything for granted in our codebase, inquire into any suspicious thing.

File: src/tribler-core/tribler_core/modules/metadata_store/gen_test_data.py [hunk 1]
```diff
@@ -16,9 +17,8 @@
     CHANNEL_TORRENT_UPDATED,
 )
 from tribler_core.tests.tools.common import TORRENT_UBUNTU_FILE, TORRENT_VIDEO_FILE
-from tribler_core.utilities.path_util import Path
 
-DATA_DIR = Path(__file__).parent / '..' / '..' / 'data'
+DATA_DIR = pathlib.Path(pathlib.Path(__file__).parent, '..', '..', 'data')
```
Comment (ichorid) [inline src/tribler-core/tribler_core/modules/metadata_store/gen_test_data.py]: Just change it to `TESTS_DATA_DIR`

File: src/tribler-core/tribler_core/modules/metadata_store/restapi/channels_endpoint.py [hunk 2]
```diff
@@ -246,7 +245,7 @@ def get_channel_from_request(self, request):
         torrents_dir = None
         if parameters.get('torrents_dir', None):
             torrents_dir = parameters['torrents_dir']
```
Comment (ichorid) [inline src/tribler-core/tribler_core/modules/metadata_store/restapi/channels_endpoint.py]: It is better to make `torrents_dir` a Path object itself, and then operate on that.

File: src/tribler-core/tribler_core/modules/metadata_store/tests/test_channel_metadata.py [hunk 3]
```diff
@@ -337,7 +335,7 @@ def generate_channel(recurse=False, status=NEW):
         # Remove the channel and read it back from disk
         for c in chan.contents:
             c.delete()
-        my_dir = path_util.abspath(self.mds.ChannelMetadata._channels_dir / chan.dirname)
+        my_dir = pathlib.Path(self.mds.ChannelMetadata._channels_dir, chan.dirname).resolve()
```
Comment (ichorid) [inline src/tribler-core/tribler_core/modules/metadata_store/tests/test_channel_metadata.py]: Is `resolve` really necessary? `/` syntax?

File: src/tribler-core/tribler_core/modules/process_checker.py [hunk 4]
```diff
@@ -56,7 +56,7 @@ def create_lock_file(self):
         might be called before the .Tribler directory has been created.
         """
         if not self.state_directory.exists():
-            path_util.makedirs(self.state_directory)
+            pathlib.Path(self.state_directory).mkdir(parents=True)
```
Comment (ichorid) [inline src/tribler-core/tribler_core/modules/process_checker.py]: Try making `self.state_directory` into a Path object in `__init__` and where it is set/used.

File: src/tribler-core/tribler_core/modules/resource_monitor.py [hunk 5]
```diff
@@ -167,14 +166,13 @@ def write_resource_logs(self, time_seconds):
...
             return
         with self.resource_log_file.open(mode="a+") as output_file:
-            output_file.write(u"%s, %s, %s\n" % (time_seconds,
-                                                self.memory_data[len(self.memory_data)-1][1],
-                                                self.cpu_data[len(self.cpu_data)-1][1]))
+            output_file.write(f"{time_seconds}, {self.memory_data[len(self.memory_data)-1][1]}, "
+                                f"{self.cpu_data[len(self.cpu_data)-1][1]}\n")
 
     def reset_resource_logs(self):
         resource_dir = self.resource_log_file.parent
```
Comment (ichorid) [inline src/tribler-core/tribler_core/modules/resource_monitor.py]: Make `resource_dir` a Path object?

File: src/tribler-core/tribler_core/modules/tracker_manager.py [hunk 6]
```diff
@@ -126,12 +126,20 @@ def get_next_tracker_for_auto_check(self):
         Gets the next tracker for automatic tracker-checking.
         :return: The next tracker for automatic tracker-checking.
         """
-        tracker = self.tracker_store.select(lambda g: str(g.url) not in [u'no-DHT', u'DHT']
-                                            and g.alive
-                                            and g.last_check + TRACKER_RETRY_INTERVAL <= int(time.time())
-                                            and str(g.url) not in self.blacklist)\
-            .order_by(self.tracker_store.last_check).limit(1)
+        tracker = (
```
Comment (ichorid) [inline src/tribler-core/tribler_core/modules/tracker_manager.py]: Be *very* careful with PonyORM queries! Please, check twice that this thing works as it should! Do not make the change, if the method is not tested.

File: src/tribler-core/tribler_core/upgrade/config_converter.py [hunk 7]
```diff
@@ -92,17 +93,17 @@ def add_tribler_config(new_config, old_config):
             temp_config = config.copy()
             if section == "Tribler" and name == "default_anonymity_enabled":
                 temp_config.set_default_anonymity_enabled(value)
-            if section == "Tribler" and name == "default_number_hops":
+            elif section == "Tribler" and name == "default_number_hops":
                 temp_config.set_default_number_hops(value)
-            if section == "downloadconfig" and name == "saveas":
+            elif section == "downloadconfig" and name == "saveas":
```
Comment (ichorid) [inline src/tribler-core/tribler_core/upgrade/config_converter.py]: nest "downloadconfig" check?

File: src/tribler-core/tribler_core/utilities/osutils.py [hunk 8]
```diff
@@ -119,16 +116,18 @@ def get_free_space(path):
 
     if sys.platform == 'win32':
         from win32file import GetDiskFreeSpaceEx
-        return GetDiskFreeSpaceEx(path_util.splitdrive(path_util.abspath(path))[0])[0]
-    else:
-        data = os.statvfs(path.encode("utf-8"))
-        return data.f_bavail * data.f_frsize
+        return GetDiskFreeSpaceEx(path_util.splitdrive(pathlib.Path(path).resolve())[0])[0]
```
Comment (ichorid) [inline src/tribler-core/tribler_core/utilities/osutils.py]: Just delete this :hankey: ...

Comment (Solomon1732) [reply src/tribler-core/tribler_core/utilities/osutils.py]: Deleted.

File: src/integration_tests/test_live_downloads.py [hunk 9]
```diff
@@ -125,11 +127,11 @@ def screenshot(self, widget, name=None):
...
 
         self.screenshots_taken += 1
-        img_name = 'screenshot_%d.jpg' % self.screenshots_taken
+        img_name = f'screenshot_{self.screenshots_taken}.jpg'
         if name is not None:
-            img_name = 'screenshot_%s.jpg' % name
+            img_name = f'screenshot_{name}.jpg'
 
-        screenshots_dir = Path(tribler_gui.__file__).parent / 'screenshots'
+        screenshots_dir = pathlib.Path(pathlib.Path(tribler_gui.__file__).parent, 'screenshots')
```
[the remark this answers is not in the document]
Comment (Solomon1732) [reply src/integration_tests/test_live_downloads.py]: Got it 👍

File: src/integration_tests/test_live_downloads.py [hunk 10]
```diff
@@ -18,18 +20,18 @@
...
 from tribler_gui.widgets.home_recommended_item import HomeRecommendedItem
 from tribler_gui.widgets.loading_list_item import LoadingListItem
 
-default_download_dir = Path(__file__).parent / u"Downloads"
-default_output_file = Path(__file__).parent / u"output.csv"
-default_state_dir = Path(__file__).parent / u".Tribler"
-default_timeout = 60000  # milliseconds
+_FILE_PARENT_DIR = pathlib.Path(__file__).parent
+default_download_dir = pathlib.Path(_FILE_PARENT_DIR, "Downloads")
+default_output_file = pathlib.Path(_FILE_PARENT_DIR, "output.csv")
```
[the remark this answers is not in the document]
Comment (Solomon1732) [reply src/integration_tests/test_live_downloads.py]: Oh, ok. Gotcha

File: src/tribler-core/tribler_core/modules/metadata_store/gen_test_data.py (hunk 1 above)
Comment (Solomon1732) [reply src/tribler-core/tribler_core/modules/metadata_store/gen_test_data.py]: Done.

File: src/tribler-core/tribler_core/modules/tracker_manager.py (hunk 6 above)
Comment (Solomon1732) [reply src/tribler-core/tribler_core/modules/tracker_manager.py]: Will undo the change.

Comment (Solomon1732) [reply src/tribler-core/tribler_core/modules/tracker_manager.py]: Undid the change.

File: src/integration_tests/test_live_downloads.py [hunk 11]
```diff
@@ -54,7 +56,7 @@
     # Set up logging before starting the GUI
     setup_gui_logging()
 
-    core_script_file = path_util.abspath(run_tribler.__file__)
+    core_script_file = pathlib.Path(run_tribler.__file__).resolve()
```
[the remark this answers is not in the document]
Comment (Solomon1732) [reply src/integration_tests/test_live_downloads.py]: Ok. Undid change to the line.

File: src/tribler-core/tribler_core/__init__.py [hunk 12]
```diff
@@ -2,14 +2,13 @@
...
 import sys
 
-from tribler_core.utilities.path_util import Path
-
-dir_path = Path(__file__).parent
+dir_path = pathlib.Path(__file__).parent
 
 # Make sure AnyDex can be imported
-sys.path.insert(1, os.path.join(dir_path, "anydex"))
+sys.path.insert(1, pathlib.Path(dir_path, "anydex"))
```
[the remark this answers is not in the document]
Comment (Solomon1732) [reply src/tribler-core/tribler_core/__init__.py]: Done.

File: src/tribler-core/tribler_core/upgrade/config_converter.py (hunk 7 above)
Comment (Solomon1732) [reply src/tribler-core/tribler_core/upgrade/config_converter.py]: Done.

File: src/tribler-core/tribler_core/modules/metadata_store/tests/test_channel_metadata.py (hunk 3 above)
Comment (Solomon1732) [reply src/tribler-core/tribler_core/modules/metadata_store/tests/test_channel_metadata.py]: Removed it.

File: src/tribler-core/tribler_core/modules/process_checker.py (hunk 4 above)
Comment (Solomon1732) [reply src/tribler-core/tribler_core/modules/process_checker.py]: Done

File: src/tribler-core/tribler_core/modules/resource_monitor.py (hunk 5 above)
Comment (Solomon1732) [reply src/tribler-core/tribler_core/modules/resource_monitor.py]: In this case it's unnecessary as `Path.parent` returns a `Path` object.
https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.parent

File: src/tribler-core/tribler_core/modules/metadata_store/restapi/channels_endpoint.py (hunk 2 above)
Comment (Solomon1732) [reply src/tribler-core/tribler_core/modules/metadata_store/restapi/channels_endpoint.py]: Done

**Comment by Solomon1732** [comment]
I will go through it later (right now it's late for me) and clean this up. It will include (but won't be limited to) squashing commits, fixing build failures, and using `/` syntax instead of constructing a `Path` object.

**Comment by Solomon1732** [comment]
```
8: E0401: (import-error), : Unable to import 'tribler_core.test.tools.common'
```
Odd. Why? Time to dig in.

**Comment by Solomon1732** [comment]
```
======================================================================
ERROR: Test committing entries into several squashed blobs and reading them back
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\users\heteoma\appdata\local\programs\python\python37\lib\site-packages\asynctest\case.py", line 297, in run
    self._run_test_method(testMethod)
  File "c:\users\heteoma\appdata\local\programs\python\python37\lib\site-packages\asynctest\case.py", line 352, in _run_test_method
    result = method()
  File "C:\workspace\jenkins2\workspace\GH_Tribler_PR_Tests\PR_async_win64\tribler\src\tribler-core\tribler_core\tests\tools\test_as_server.py", line 53, in check
    result = fun(*argv, **kwargs)
  File "<string>", line 2, in test_multiple_squashed_commit_and_read
  File "c:\users\heteoma\appdata\local\programs\python\python37\lib\site-packages\pony\orm\core.py", line 528, in new_func
    result = func(*args, **kwargs)
  File "C:\workspace\jenkins2\workspace\GH_Tribler_PR_Tests\PR_async_win64\tribler\src\tribler-core\tribler_core\modules\metadata_store\tests\test_store.py", line 181, in test_multiple_squashed_commit_and_read
    self.mds.process_channel_dir(channel_dir, channel.public_key, channel.id_, skip_personal_metadata_payload=False)
  File "C:\workspace\jenkins2\workspace\GH_Tribler_PR_Tests\PR_async_win64\tribler\src\tribler-core\tribler_core\modules\metadata_store\store.py", line 241, in process_channel_dir
    blob_sequence_number = int(filename[: -len(str(BLOB_EXTENSION))])
ValueError: invalid literal for int() with base 10: ''
-------------------- >> begin captured logging << --------------------
DEBUG   1578407727 selector_events:asyncio:53   Using selector: SelectSelector
INFO    1578407727 channel_metadata:MetadataStore:282  Channel 91413fb03afc507f3732c794b4a793dbc084c6154cc9754697528686de292469e4c290fb7043d7c23d75cb96c7ad49f89d9087678f2eceb0a17e3672cb8bab05 committed with 10 new entries. New version is 1578407727013
DEBUG   1578407727           store:MetadataStore:233  Starting processing channel dir C:\workspace\jenkins2\workspace\GH_Tribler_PR_Tests\PR_async_win64\tribler\src\tribler-core\tribler_core\tests\tools\temp\TestMetadataStore_tribler_test_session_C7LMSC\91413fb03afc507f3732c794b4a793db22433ed90d67b366. Channel 91413fb03afc507f3732c794b4a793dbc084c6154cc9754697528686de292469e4c290fb7043d7c23d75cb96c7ad49f89d9087678f2eceb0a17e3672cb8bab05 local/max version 0/1578407727013
--------------------- >> end captured logging << ---------------------
```
Not sure whether it's because of what I changed there or because of the test...
https://jenkins-ci.tribler.org/job/GH_Tribler_PR_Tests/job/PR_async_win64/71/artifact/output/test_runners_output/2.err/*view*/
Key lines:
```
File "C:\workspace\jenkins2\workspace\GH_Tribler_PR_Tests\PR_async_win64\tribler\src\tribler-core\tribler_core\modules\metadata_store\tests\test_store.py", line 181, in test_multiple_squashed_commit_and_read
    self.mds.process_channel_dir(channel_dir, channel.public_key, channel.id_, skip_personal_metadata_payload=False)
  File "C:\workspace\jenkins2\workspace\GH_Tribler_PR_Tests\PR_async_win64\tribler\src\tribler-core\tribler_core\modules\metadata_store\store.py", line 241, in process_channel_dir
    blob_sequence_number = int(filename[: -len(str(BLOB_EXTENSION))])
```

**Comment by Solomon1732** [comment]
> 8: E0401: (import-error), : Unable to import 'tribler_core.test.tools.common'

It's test**s**, not test.

**Comment by Solomon1732** [comment]
https://jenkins-ci.tribler.org/job/GH_Tribler_PR_Tests/job/PR_async_win64/81/console

```
HTTP ERROR 403

Problem accessing /job/GH_Tribler_PR_Tests/job/PR_async_win64/81/logText/progressiveHtml. Reason:

> No valid crumb was included in the request
```
It's not because of my changes, is it?

File: src/tribler-core/tribler_core/upgrade/tests/test_config_upgrade_70_71.py [hunk 13]
```diff
@@ -33,15 +33,15 @@ def test_read_test_libtribler_conf(self):
...
+        os.environ['TSTATEDIR'] = str(self.session_base_dir)
         old_config = RawConfigParser()
-        old_config.read(self.CONFIG_PATH / "libtribler70.conf")
+        old_config.read(pathlib.Path(self.CONFIG_PATH, "libtribler70.conf"))
         new_config = TriblerConfig()
         result_config = add_libtribler_config(new_config, old_config)
         self.assertEqual(result_config.get_tunnel_community_socks5_listen_ports(), [1, 2, 3, 4, 5, 6])
         self.assertEqual(result_config.get_anon_proxy_settings(), (2, ("127.0.0.1", [5, 4, 3, 2, 1]), ''))
         self.assertEqual(result_config.get_credit_mining_sources(), ['source1', 'source2'])
-        self.assertEqual(result_config.get_log_dir(), path_util.Path('/a/b/c').absolute())
```
Comment (Solomon1732) [inline src/tribler-core/tribler_core/upgrade/tests/test_config_upgrade_70_71.py]: What's this `absolute` method? Is it suppose to be `resolve` or `normpath` or something? For `pathlib.Path` and `path_util.Path` respectively.

https://github.com/Tribler/tribler/blob/devel/src/tribler-core/tribler_core/utilities/path_util.py
https://docs.python.org/3/library/pathlib.html#pathlib.Path.resolve

**Comment by Solomon1732** [comment]
Can this be retested? Some of the tests did not execute due to timeout for some reason. It doesn't seem to happen for every push I made, rather it's sporadic somewhat 😅

**Comment by Solomon1732** [comment]
From https://github.com/Tribler/tribler/blob/devel/src/tribler-core/tribler_core/modules/libtorrent/libtorrent_download_impl.py#L275
```python
def set_vod_mode(self, enable=True):

    self._logger.debug("LibtorrentDownloadImpl: set_vod_mode for %s (enable = %s)", self.tdef.get_name(), enable)

    if enable:
        self.vod_seekpos = 0
        self.vod_index = self.config.get_selected_files()[0] if self.tdef.is_multifile_torrent() else 0
    # More code

def get_vod_fileindex(self):
    if self.vod_index is not None:
        return self.vod_index
    return -1

@check_handle(0)
def get_vod_filesize(self):
    fileindex = self.get_vod_fileindex()
    torrent_info = get_info_from_handle(self.handle)
    if fileindex >= 0 and torrent_info:
        file_entry = torrent_info.file_at(fileindex)
        return file_entry.size
    return 0
```
From https://github.com/Tribler/tribler/blob/devel/src/tribler-core/tribler_core/modules/libtorrent/download_config.py#L141
```python
def get_selected_files(self):
    """ Returns the list of files selected for download.
    @return A list of file indexes. """
    return self.config['download_defaults']['selected_file_indexes']
```
It looks like `self.vod_index = self.config.get_selected_files()[0]` returns a list of files, and then later on treats them as `int`s in `if fileindex >= 0 and torrent_info:`, which then expectedly fails. Am I getting this wrong?

**Comment by Solomon1732** [comment]
From `tribler/src/tribler-core/tribler_core/modules/libtorrent/tests/test_downloadconfig.py`:
https://github.com/Tribler/tribler/blob/devel/src/tribler-core/tribler_core/modules/libtorrent/tests/test_libtorrent_download_impl.py#L228
`self.libtorrent_download_impl.set_selected_files(['a'])`
https://github.com/Tribler/tribler/blob/devel/src/tribler-core/tribler_core/modules/libtorrent/tests/test_libtorrent_download_impl.py#L257
`self.libtorrent_download_impl.set_selected_files(['a'])`
Looks like it sets the files for strings instead of indexes. On [this](https://github.com/Tribler/tribler/blob/devel/src/tribler-core/tribler_core/modules/libtorrent/tests/test_downloadconfig.py#L46) test it checks that it raises `ValueError` because of it, while [this](https://github.com/Tribler/tribler/blob/devel/src/tribler-core/tribler_core/modules/libtorrent/tests/test_libtorrent_download_impl.py#L208) test, [this](https://github.com/Tribler/tribler/blob/devel/src/tribler-core/tribler_core/modules/libtorrent/tests/test_libtorrent_download_impl.py#L235) one, and [this](https://github.com/Tribler/tribler/blob/devel/src/tribler-core/tribler_core/modules/video_server.py#L156) module treat it as if it accepts file names instead of indexes.

In `libtorrent_mgr_impl` [this](https://github.com/Tribler/tribler/blob/devel/src/tribler-core/tribler_core/modules/libtorrent/libtorrent_download_impl.py#L618) is the code:
```python
@check_handle()
def set_selected_files(self, selected_files=None):
    if not isinstance(self.tdef, TorrentDefNoMetainfo) and not self.get_share_mode():
        if selected_files is None:
            selected_files = self.config.get_selected_files()
        else:
            self.config.set_selected_files(selected_files)
```
In other words it looks like it sets the file indexes as strings, which later on get called as `int`s, which then raise `ValueError`

**Comment by Solomon1732** [comment]
https://github.com/Tribler/tribler/blob/devel/src/tribler-core/tribler_core/modules/video_server.py#L156
`download.set_selected_files([filename])` if I'm not mistaken, it should be `fileindex` instead of `filename` in that line.

**Comment by ichorid** [comment]
retest this please

File: src/tribler-core/tribler_core/upgrade/tests/test_config_upgrade_70_71.py (hunk 13 above)
Comment (ichorid) [reply src/tribler-core/tribler_core/upgrade/tests/test_config_upgrade_70_71.py]: absolute means resolve a relative path like `../foo/bar/` to something like `/tmp/foo/bar`. It should be used sparingly, mostly to save system-dependent paths.

Comment (Solomon1732) [reply src/tribler-core/tribler_core/upgrade/tests/test_config_upgrade_70_71.py]: Ah. Gotcha. Is it alright in this case to replace `absolute` with `resolve`?

Comment (ichorid) [reply src/tribler-core/tribler_core/upgrade/tests/test_config_upgrade_70_71.py]: Yes. But try to remove it at all.

**Comment by egbertbouman** [comment]
@Solomon1732 Recently, we changed `set_selected_files`/`get_selected_files` to use indexes instead. Unfortunately, some of the tests don't seem to reflect this yet. So, the tests you are referring to should do something like:
```self.libtorrent_download_impl.set_selected_files([0])```

You're right https://github.com/Tribler/tribler/blob/devel/src/tribler-core/tribler_core/modules/video_server.py#L156 should use the file index:
```download.set_selected_files([fileindex])```

To avoid future confusion, maybe you could rename `set_selected_files`/`get_selected_files` to something like `set_selected_file_indexes`/`get_selected_file_indexes`?

File: src/tribler-core/tribler_core/upgrade/tests/test_config_upgrade_70_71.py (hunk 13 above)
Comment (Solomon1732) [reply src/tribler-core/tribler_core/upgrade/tests/test_config_upgrade_70_71.py]: Remove `test_read_test_tribler_conf`, or the specific assertion only?

**Comment by Solomon1732** [comment]
Seems like only some of the Windows tests keep timing out for some reason...

**Comment by ichorid** [comment]
Like, there are about 40 tests failing for reasons like `Is a directory`...

**Comment by Solomon1732** [comment]
I wanted to deal with the low-hanging fruit first. I'll deal with this instead, however 👍

**Comment by Solomon1732** [comment]
Added logging to the line that throws the exception. Hopefully it will show the culprit.

**Comment by Solomon1732** [comment]
```
DEBUG   1578499639  libtorrent_mgr:LibtorrentMgr:214  file opened: '/home/jenkins/ci/workspace/GH_Tribler_PR_Tests/PR_async_linux/tribler/src/tribler-core/tribler_core/tests/tools/temp/TestDownloadsEndpoint_tribler_test_session_3YKGFT/dot.Tribler/lt.state'
INFO    1578499639  libtorrent_mgr:LibtorrentMgr:222  could not load libtorrent state, got exception: FileNotFoundError(2, 'No such file or directory'). starting from scratch
```
It can't fine this file for some reason. Now to find out why…
https://jenkins-ci.tribler.org/job/GH_Tribler_PR_Tests/job/PR_async_linux/100/testReport/junit/test_downloads_endpoint/TestDownloadsEndpoint/test_load_checkpoint/

**Comment by Solomon1732** [comment]
What is responsible for creating the files declared here?
https://github.com/Solomon1732/tribler/blob/f_clean_path_util/src/tribler-core/tribler_core/tests/tools/test_as_server.py#L123

File: src/tribler-core/tribler_core/upgrade/upgrade.py [hunk 14]
```diff
@@ -186,7 +186,7 @@ def backup_state_directory(self):
                     if backup_dir in src_sub_dirs:
                         dir_copy(src_state_dir / backup_dir, dest_state_dir / backup_dir)
                     else:
-                        os.makedirs(dest_state_dir / backup_dir)
+                        (dest_state_dir / backup_dir).mkrdir(parents=True)
```
Comment (ichorid) [inline src/tribler-core/tribler_core/upgrade/upgrade.py]: `mkRdir` really?

File: src/tribler-core/tribler_core/upgrade/config_converter.py [hunk 15]
```diff
@@ -90,20 +90,13 @@ def add_tribler_config(new_config, old_config):
...
             temp_config = config.copy()
-            if section == "Tribler" and name == "default_anonymity_enabled":
-                temp_config.set_default_anonymity_enabled(value)
-            if section == "Tribler" and name == "default_number_hops":
-                temp_config.set_default_number_hops(value)
-            if section == "downloadconfig" and name == "saveas":
-                temp_config.config["download_defaults"]["saveas"] = value
-            if section == "downloadconfig" and name == "seeding_mode":
-                temp_config.config["download_defaults"]["seeding_mode"] = value
-            if section == "downloadconfig" and name == "seeding_ratio":
```
Comment (ichorid) [inline src/tribler-core/tribler_core/upgrade/config_converter.py]: where did `seeding_ratio` gone?

File: src/tribler-core/tribler_core/modules/watch_folder.py [hunk 16]
```diff
@@ -41,14 +41,11 @@ def check_watch_folder(self):
...
+        watch_dir = str(self.session.config.get_watch_folder_path())
 
         for root, _, files in os.walk(watch_dir):
-            root = path_util.Path(root)
-            for name in files:
-                if not name.endswith(".torrent"):
-                    continue
-
+            root = Path(root)
+            for name in (name for name in files if name.endswith(".torrent")):
```
Comment (ichorid) [inline src/tribler-core/tribler_core/modules/watch_folder.py]: That's a *bad* idea. As you can see, `files` is produced by the `os.walk` **iterator**. You don't really want to build intermediate tuples with that, as these are gonna be HUGE.

File: src/tribler-core/tribler_core/modules/libtorrent/torrentdef.py [hunk 17]
```diff
@@ -349,19 +348,20 @@ def _get_all_files_as_unicode_with_length(self):
...
 
                 if b"path" in file_dict:
-                    # Try to use the 'encoding' field.  If it exists,
+                    # Try to use the 'encoding' field. If it exists,
                     # it should contain something like 'utf-8'
                     if b"encoding" in self.metainfo:
                         encoding = ensure_unicode(self.metainfo[b"encoding"], "utf8")
                         try:
-                            yield (join(*[ensure_unicode(element, encoding) for element in file_dict[b"path"]]),
+                            yield (Path(*(ensure_unicode(element, encoding) for element in file_dict[b"path"])).name,
```
Comment (ichorid) [inline src/tribler-core/tribler_core/modules/libtorrent/torrentdef.py]: The original line does not seem to get the *last* part of the path, as you do with `.name` there.

**Comment by ichorid** [comment]
HINT: the smaller the PR, the easier it is to make it pass the tests and review it :wink: 
Also, splitting a PR into several smaller PRs is a nice excersise in `git` stuff as well...

**Comment by Solomon1732** [comment]
I'm closing this PR in favor of opening a set of smaller ones
